src/scenes/CreateEvent/store.js

```diff
@@ -0,0 +1,397 @@
+let TIME_GAP_MIN_SEC = 30 * 60;
+if (process.env.REACT_APP_ENV === 'dev') {
+  TIME_GAP_MIN_SEC = 2 * 60;
+}
+
+const INIT = {
+  escrowAmount: '', // type: number
+  averageBlockTime: '',
+  txFees: [],
+  txid: '',
+  isOpen: false,
+  txConfirmDialogOpen: false,
+  txSentDialogOpen: false,
+  resultSetterDialogOpen: false,
+  title: '',
+  creator: '',
+  prediction: {
+    startTime: nowPlus(15),
+    endTime: nowPlus(45),
+  },
+  resultSetting: {
+    startTime: nowPlus(45),
+    endTime: nowPlus(75),
+  },
+  outcomes: ['', ''],
+  resultSetter: '',
+  // if one of these in error is set, the form
+  // field will display the associated error message
+  error: {
+    title: '',
+    creator: '',
+    prediction: {
+      startTime: '',
+      endTime: '',
+    },
+    resultSetting: {
+      startTime: '',
+      endTime: '',
+    },
+    outcomes: ['', ''],
+    resultSetter: '',
+  },
+};
+
+
+export default class CreateEventStore {
+  escrowAmount = INIT.escrowAmount
+  averageBlockTime = INIT.averageBlockTime // TODO: maybe move to global store?
+  txFees = INIT.txFees
+  txid = INIT.txid // used in txSentDialog
+  @observable txConfirmDialogOpen = INIT.txConfirmDialogOpen
+  @observable txSentDialogOpen = INIT.txSentDialogOpen
+  @observable resultSetterDialogOpen = INIT.resultSetterDialogOpen
+  // form fields
+  @observable isOpen = INIT.isOpen
+  @observable title = INIT.title
+  @observable creator = INIT.creator // address
+  prediction = observable(INIT.prediction)
+  resultSetting = observable(INIT.resultSetting)
+  @observable outcomes = INIT.outcomes
+  @observable resultSetter = INIT.resultSetter // address
+  error = observable(INIT.error)
+  @computed get hasEnoughQtum() {
+    const totalQtum = _.sumBy(this.app.wallet.addresses, ({ qtum }) => qtum);
+    return totalQtum >= maxTransactionFee;
+  }
+  @computed get warning() {
+    if (!this.hasEnoughQtum) {
+      return {
+        id: 'str.notEnoughQtum',
+        message: 'You don\'t have enough QTUM',
+      };
+    }
+    return {};
+  }
+  @computed get blockNum() {
+    const { prediction, resultSetting } = this;
+    return {
+      prediction: {
+        startTime: this.calculateBlock(prediction.startTime),
+        endTime: this.calculateBlock(prediction.endTime),
+      },
+      resultSetting: {
+        startTime: this.calculateBlock(resultSetting.startTime),
+        endTime: this.calculateBlock(resultSetting.endTime),
+      },
+    };
+  }
+  @computed get isAllValid() {
+    const { title, creator, prediction, resultSetting, outcomes, resultSetter } = this.error;
+    return (
+      // all fields set
+      this.title
+      && this.outcomes.every(outcome => !!outcome)
+      && this.resultSetter
+      // no errors
+      && !title
+      && !creator
+      && !prediction.startTime
+      && !prediction.endTime
+      && !resultSetting.startTime
+      && !resultSetting.endTime
+      && outcomes.every(outcome => !outcome)
+      && !resultSetter
```

Comment:
arent these checking for undefined instead of empty?

Reply:
```js
!'' === true
!undefined === true
!null === true
```
since all these are strings, we're just checking to see if they are empty, if they are, we're good